### PR TITLE
Feat feed functionality

### PR DESCRIPTION
**Pull request by coldpigli**
1) Completely redesigned the app
2) Added reducers for auth and posts
3) Added functionality to the feed page.
4) Made the app responsive

### CODE REVIEW

File: src/components/Feedlisting/Feedlisting.jsx [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+import { getAllPosts } from "services";
 
 const Feedlisting = () => {
+
+  const {userData} = useSelector((state)=>state.auth);
+  const {postList} = useSelector((state)=>state.posts);
+  const dispatch = useDispatch();
+
+  useEffect(()=>{
+    dispatch(getAllPosts())
```
Comment (vaibhav-mahalle) [inline src/components/Feedlisting/Feedlisting.jsx:14]: liked the use of getAllPosts()

File: src/components/Feedlisting/Feedlisting.jsx [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
       <HStack justify="space-between" mb="1rem">
         <Heading size="md">Scroll</Heading>
         <Text fontSize="sm">Showing Recent</Text>
       </HStack>
       <VStack align="stretch" spacing="6" borderRadius='1rem'>
-        <PostItem/>
-        <PostItem/>
-        <PostItem/>
-        <PostItem/>
+        {
```
Comment (vaibhav-mahalle) [inline src/components/Feedlisting/Feedlisting.jsx:24]: well-optimized using map function.

**Review by vaibhav-mahalle — APPROVED** [review]
liked the use of custom hooks and very clean code

File: src/services/postServices.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import axios from "axios";
+
+const getAllPosts = createAsyncThunk("posts/getAllPosts",
+async ()=>{
+    try{
+        const res = await axios.get('/api/posts');
+        console.log("I was here for posts")
+        return res.data;
+    }catch(err){
+        console.log(err);
```
Comment (rahulcs754) [inline src/services/postServices.js:11]: I think you should remove console.log for error handling you should use createAsyncThunk api

**Review by rahulcs754 — APPROVED** [review]
all Code looks clean and good

**Pull request merged**